server/api.go

```diff
@@ -167,6 +185,28 @@ func (a *apiServer) Publish(ctx context.Context, req *client.PublishRequest) (
 	}
 	a.logger.Debugf("api: Publish [subject=%s]", subject)
 
+	stream := a.metadata.GetStream(req.Stream)
+	if stream == nil {
+		return nil, status.Error(codes.NotFound, fmt.Sprintf("No such stream: %s", req.Stream))
+	}
+	partition, ok := stream.partitions[0] // Only check the first partition for now
```

Comment:
So have we decided what the policy should be here? Do all partitions get paused simultaneously? Or they can independently pause and resume?

Reply:
We could add an option to perform either. Internally the stream does not store anything, so the pause status is stored into a partition struct. For now the API pauses the whole stream, that means all partitions. When one partition gets resumed, all other partitions from the stream also get resumed. I think we could see that Tyler says about this, unless you have some idea on what we should do?

Reply:
ok